### PR TITLE
feat(dlm):support viewing task logs

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
Support viewing task logs to help customers better understand task status and effectively  troubleshoot problems.
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

The SNAPSHOT version currently in use needs to be updated after the official version is released.

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/DLMConfiguration.java [hunk 1]
```diff
@@ -44,13 +45,25 @@ public class DLMConfiguration {
     @Value("${odc.task.dlm.single-task-thread-pool-size:15}")
     private int singleTaskThreadPoolSize;
 
+    @Value("${odc.task.dlm.task-connection-query-timeout-seconds:60}")
+    private int taskConnectionQueryTimeout;
+
+    @Value("${odc.task.dlm.sharding-strategy:MATCH}")
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/DLMConfiguration.java:51]: The default value is not the same as your value in config_system_configuration.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/utils/DataArchiveConditionUtil.java [hunk 2]
```diff
@@ -58,6 +59,9 @@ private static String replaceVariables(String condition, Map<String, String> var
     }
 
     private static Map<String, String> getVariablesMap(List<OffsetConfig> variables) {
+        if (variables == null || variables.isEmpty()) {
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/utils/DataArchiveConditionUtil.java:62]: may simplify to `CollectionUtils.isEmpty(variables)`

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/flowtask/DataArchivePreprocessor.java [hunk 3]
```diff
@@ -104,6 +114,13 @@ public void process(CreateFlowInstanceReq req) {
             if (parameters.getOperationType() == OperationType.UPDATE) {
                 parameters.setDescription(req.getDescription());
                 ScheduleEntity scheduleEntity = scheduleService.nullSafeGetById(parameters.getTaskId());
+                List<ScheduleTaskEntity> runningTasks = scheduleTaskService.listTaskByJobNameAndStatus(
+                        scheduleEntity.getId().toString(), Arrays.asList(TaskStatus.RUNNING, TaskStatus.PREPARING));
+                if (scheduleEntity.getStatus() != ScheduleStatus.PAUSE || !runningTasks.isEmpty()) {
+                    throw new IllegalStateException(
+                            String.format("Resume schedule not allowed,schedule status=%s,running tasks=%s",
+                                    scheduleEntity.getStatus(), runningTasks.size()));
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/flowtask/DataArchivePreprocessor.java:122]: `runningTasks.size()` is not compatible with your log `running tasks=%s`

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/flowtask/DataArchivePreprocessor.java [hunk 4]
```diff
@@ -104,6 +114,13 @@ public void process(CreateFlowInstanceReq req) {
             if (parameters.getOperationType() == OperationType.UPDATE) {
                 parameters.setDescription(req.getDescription());
                 ScheduleEntity scheduleEntity = scheduleService.nullSafeGetById(parameters.getTaskId());
+                List<ScheduleTaskEntity> runningTasks = scheduleTaskService.listTaskByJobNameAndStatus(
+                        scheduleEntity.getId().toString(), Arrays.asList(TaskStatus.RUNNING, TaskStatus.PREPARING));
+                if (scheduleEntity.getStatus() != ScheduleStatus.PAUSE || !runningTasks.isEmpty()) {
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/flowtask/DataArchivePreprocessor.java:119]: What's the relation between this condition `scheduleEntity.getStatus() != ScheduleStatus.PAUSE || !runningTasks.isEmpty()` and your log 'Resume schedule not allowed ...'

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/utils/DataArchiveConditionUtil.java (hunk 2 above)
Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/utils/DataArchiveConditionUtil.java]: done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/flowtask/DataArchivePreprocessor.java (hunk 4 above)
Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/flowtask/DataArchivePreprocessor.java:119]: In order to ensure scheduling security, editing is only allowed when the plan is paused and there are no subtasks executing.

File: server/odc-server/src/main/resources/data.sql [hunk 5]
```diff
@@ -684,6 +684,9 @@ INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES
 INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES( 'odc.task.dlm.single-task-thread-pool-size', '12', 'DLM 单个任务可用线程数' ) ON DUPLICATE KEY UPDATE `id` = `id`;
 INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES( 'odc.task.dlm.thread-pool-size', '100', '单个 POD 中 DLM 任务线程池大小' ) ON DUPLICATE KEY UPDATE `id` = `id`;
 INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES( 'odc.task.dlm.support-breakpoint-recovery', 'true', 'DLM 任务是否开启断点恢复' ) ON DUPLICATE KEY UPDATE `id` = `id`;
+INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES( 'odc.task.dlm.task-connection-query-timeout-seconds', '180', 'DLM 任务 SQL 超时时间' ) ON DUPLICATE KEY UPDATE `id` = `id`;
+INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES( 'odc.task.dlm.sharding-strategy', 'MATCH', 'DLM 分片策略，默认值 FIXED_LENGTH 表示均匀分片，适合小规格实例。使用 MATCH 策略时将出现少量慢 SQL，整体性能会有较大提升，适合大规格实例。' ) ON DUPLICATE KEY UPDATE `id` = `id`;
```
Comment (MarkPotato777) [inline server/odc-server/src/main/resources/data.sql:688]: the description does not match your value, p.s. recommend in English :)

Comment (guowl3) [reply server/odc-server/src/main/resources/data.sql]: changed the default strategy to FIX_LENGTH

**Comment by MarkPotato777** [comment]
LGTM

**Review by MarkPotato777 — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**